File: LeetCode/190_颠倒二进制位.java
public class Solution {
    // you need treat n as an unsigned value
    public int reverseBits(int n) {
        int res = 0;
        for(int i = 0; i < 32; i++) {
            int afterAnd = (n >> i) & 1;
            res = (res << 1) + afterAnd;
        }

        return res;
    }
}

File: LeetCode/190_颠倒二进制位.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t res = 0;
        int k = 1;
        while(k <= 32) {
            res = res * 2 + n % 2;
            n /= 2;
            k++;
        }
        
        return res;
    }
};


class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t res = 0;
        int k = 32;
        while(k--){
            res <<= 1; // 左移一位
            res += n & 1;
            n >>= 1; // 右移一位
        }
        return res;
    }
};


